staging/src/k8s.io/component-helpers/kubernetesx509/kubernetesx509.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package kubernetesx509 contains routines for creating and parsing
+// x509 certificates and certificate requests that embed Kubernetes-specific
+// X.509 extensions that communicate the Kubernetes identity of a given
+// workload.
+package kubernetesx509
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"fmt"
+)
+
+var (
+	// The ASN.1 Private Enterprise Number assigned to the Kubernetes project.
+	// Used to name X.509 extensions that communicate Kubernetes-specific
+	// concepts.
+	KubernetesPEN = asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 57683}
+
+	// OIDPodIdentity identifies the Kubernetes PodIdentity X.509 extension.
+	OIDPodIdentity = makeKubernetesOID(1)
+)
+
+func makeKubernetesOID(subIDs ...int) asn1.ObjectIdentifier {
+	base := asn1.ObjectIdentifier{}
+	base = append(base, KubernetesPEN...)
+	base = append(base, subIDs...)
+	return base
+}
```

Comment:
I think the idea of a builder, but we need to put at least the oid bits of this in a single place so we don't accidentally duplicate oids. Since there's also no way to make a package-public constant version of this, I'd suggest a getter function that returns a new instance, so badly behaved things can't mutate these public package variables.

See https://github.com/kubernetes/kubernetes/pull/127897#discussion_r1833399650 as a suggested location / approach.

I think having the asn1 serialization of this be defined in that package is probably also a good idea (so the pkix.Extension definition and the podIdentityASN1 type)

Reply:
OK, this is not going to make the code freeze, so I will wait for #127897 to create that file and then integrate there.